File: DotNET/Endpoint Examples/Multipart Payload/blank-pdf.cs
/*
 * What this sample does:
 * - Calls /blank-pdf via multipart/form-data to create a three-page blank PDF.
 * - Routed from Program.cs as: `dotnet run -- blank-pdf-multipart`.
 *
 * Setup (environment):
 * - Copy .env.example to .env
 * - Set PDFREST_API_KEY=your_api_key_here
 * - Optional: set PDFREST_URL to override the API region. For EU/GDPR compliance and proximity, use:
 *     PDFREST_URL=https://eu-api.pdfrest.com
 *   For more information visit https://pdfrest.com/pricing#how-do-eu-gdpr-api-calls-work
 *
 * Usage:
 *   dotnet run -- blank-pdf-multipart
 *
 * Output:
 * - Prints the JSON response. Validation errors (args/env) exit non-zero.
 */

namespace Samples.EndpointExamples.MultipartPayload
{
    public static class BlankPdf
    {
        public static async Task Execute(string[] args)
        {
            var apiKey = Environment.GetEnvironmentVariable("PDFREST_API_KEY");
            if (string.IsNullOrWhiteSpace(apiKey))
            {
                Console.Error.WriteLine("Missing required environment variable: PDFREST_API_KEY");
                Environment.Exit(1);
                return;
            }

            var baseUrl = Environment.GetEnvironmentVariable("PDFREST_URL") ?? "https://api.pdfrest.com";

            using (var httpClient = new HttpClient { BaseAddress = new Uri(baseUrl) })
            using (var request = new HttpRequestMessage(HttpMethod.Post, "blank-pdf"))
            {
                request.Headers.TryAddWithoutValidation("Api-Key", apiKey);
                request.Headers.Accept.Add(new("application/json"));

                var multipartContent = new MultipartFormDataContent
                {
                    { new StringContent("letter"), "page_size" },
                    { new StringContent("3"), "page_count" },
                    { new StringContent("portrait"), "page_orientation" }
                };

                request.Content = multipartContent;
                var response = await httpClient.SendAsync(request);
                var apiResult = await response.Content.ReadAsStringAsync();

                Console.WriteLine("blank-pdf response received.");
                Console.WriteLine(apiResult);

                if (!response.IsSuccessStatusCode)
                {
                    Environment.ExitCode = 1;
                }
            }
        }
    }
}

File: Python/Endpoint Examples/Multipart Payload/blank-pdf.py
from requests_toolbelt import MultipartEncoder
import requests
import json

# By default, we use the US-based API service. This is the primary endpoint for global use.
api_url = "https://api.pdfrest.com"

# For GDPR compliance and enhanced performance for European users, you can switch to the EU-based service by uncommenting the URL below.
# For more information visit https://pdfrest.com/pricing#how-do-eu-gdpr-api-calls-work
#api_url = "https://eu-api.pdfrest.com"

blank_pdf_endpoint_url = api_url + '/blank-pdf'

# The /blank-pdf endpoint generates an empty PDF, so no file upload is required.
mp_encoder_blank_pdf = MultipartEncoder(
    fields={
        'page_size': 'letter',
        'page_count': '3',
        'page_orientation': 'portrait',
    }
)

# Let's set the headers that the blank-pdf endpoint expects.
headers = {
    'Accept': 'application/json',
    'Content-Type': mp_encoder_blank_pdf.content_type,
    'Api-Key': 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx'  # place your api key here
}

print("Sending POST request to blank-pdf endpoint...")
response = requests.post(blank_pdf_endpoint_url, data=mp_encoder_blank_pdf, headers=headers)

print("Response status code: " + str(response.status_code))

if response.ok:
    response_json = response.json()
    print(json.dumps(response_json, indent=2))
else:
    print(response.text)

# If you would like to download the file instead of getting the JSON response, please see the 'get-resource-multipart.py' sample.
